tools/benchmark/cmd/watch.go

```diff
@@ -97,6 +98,11 @@ func watchFunc(cmd *cobra.Command, args []string) {
 
 	pdoneC := printRate(results)
 
+	for i := range streams {
+		wg.Add(1)
```

Comment:
The usage of wait group is tricky here since we call wait.Done() in doWatch many times. The normal/common use case of waitgroup is to wait for a collection of goroutines to finish. 

You can find the example here https://golang.org/pkg/sync/#WaitGroup. 

I do not like the idea to use it to do other kind of sync, like counting the number of requests.